tests/e2e/swarm_test.go

```diff
@@ -0,0 +1,141 @@
+	out, err := dockercli.ListNodes(s.master)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create the volume
+	out, err = dockercli.CreateVolume(s.master, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	status := verification.VerifyDetachedStatus(s.volumeName, s.master, s.esxName)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", s.volumeName))
+}
+
+func (s *SwarmTestSuite) TearDownTest(c *C) {
+	// Clean up the volume
+	out, err := dockercli.DeleteVolume(s.master, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+}
+
+var _ = Suite(&SwarmTestSuite{})
+
+// Test vDVS usage in swarm mode
+//
+// Test steps:
+// 1. Create a docker service with replicas setting to 1
+// 2. Verify the service is up and running with one node
+// 3. Verify one container is spawned
+// 5. Verify the volume is in attached status
+// 6. Scale the service to set replica numbers to 2
+// 7. Verify the service is up and running with two nodes
+// 8. Verify 2 containers are spawned
+// 9. Stop one node of the service
+// 10. Verify the service is still running with two nodes
+// 11. Verify there are still 2 containers up and running
+// 12. Verify the volume is in attached status
+// 13. Delete the volume - expect fail
+// 14. Remove the service
+// 15. Verify the service is gone
+// 16. Verify the volume is in detached status
+func (s *SwarmTestSuite) TestDockerSwarm(c *C) {
+	log.Printf("START: swarm_test.TestDockerSwarm")
+
+	fullVolumeName := verification.GetFullVolumeName(s.master, s.volumeName)
+	opts := "--replicas 1 --mount type=volume,source=" + fullVolumeName + ",target=/vol,volume-driver=vsphere busybox tail -f /dev/null"
+	out, err := dockercli.CreateService(s.master, s.serviceName, opts)
+	c.Assert(err, IsNil, Commentf(out))
+
+	status := verification.VerifyDockerService(s.master, s.serviceName, 1)
+	c.Assert(status, Equals, true, Commentf("Service %s is not running", s.serviceName))
+
+	status, _ = verification.VerifyDockerContainer(s.dockerHosts, s.serviceName, 1)
+	c.Assert(status, Equals, true, Commentf("Container %s is not running", s.serviceName))
+
+	status = verification.VerifyAttachedStatus(s.volumeName, s.master, s.esxName)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
+
+	out, err = dockercli.ScaleService(s.master, s.serviceName, 2)
+	c.Assert(err, IsNil, Commentf(out))
+
+	status = verification.VerifyDockerService(s.master, s.serviceName, 2)
+	c.Assert(status, Equals, true, Commentf("Service %s is not running", s.serviceName))
+
+	status, host := verification.VerifyDockerContainer(s.dockerHosts, s.serviceName, 2)
+	c.Assert(status, Equals, true, Commentf("Container %s is not running on any hosts", s.serviceName))
+
+	containerName, err = verification.GetContainerName(host, s.serviceName+".1")
+	c.Assert(err, IsNil, Commentf("Failed to retrieve container name: %s", containerName))
+	out, err = dockercli.StopService(host, containerName)
```

Comment:
First of all, if we need any test to cover new scenarios, that should be done in a separate PR. Moreover, I still don't follow why we need a e2e test for #1315. We can discuss that in the issue itself instead of in this PR.